pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -305,22 +481,94 @@ func configMapDeleted(obj interface{}) {
-	// Internal FSS configmap
-	if fssConfigMap.Name == k8sOrchestratorInstance.internalFSS.configMapName && fssConfigMap.Namespace == k8sOrchestratorInstance.internalFSS.configMapNamespace {
-		for featureName := range k8sOrchestratorInstance.internalFSS.featureStates {
-			k8sOrchestratorInstance.internalFSS.featureStates[featureName] = strconv.FormatBool(false)
-		}
-		log.Infof("configMapDeleted: %v deleted. Setting internal feature state values to false %v", fssConfigMap.Name,
-			k8sOrchestratorInstance.internalFSS.featureStates)
+}
+
+// fssCRAdded adds supervisor feature state switch values from the cnscsisvfeaturestate CR
+func fssCRAdded(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	var svFSSObject featurestatesv1alpha1.CnsCsiSvFeatureStates
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &svFSSObject)
+	if err != nil {
+		log.Errorf("fssCRAdded: failed to cast object to %s. err: %v", featurestates.CRDSingular, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRAdded: Ignoring %s CR object with name %q", featurestates.CRDSingular, svFSSObject.Name)
+		return
 	}
+	setSvFssCRAvailability(true)
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(fss.Enabled)
+	}
+	log.Infof("fssCRAdded: New supervisor feature states values stored successfully from %s CR object: %v",
+		featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+}
+
+// fssCRUpdated updates supervisor feature state switch values from the cnscsisvfeaturestate CR
+func fssCRUpdated(oldObj, newObj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	var svFSSObject featurestatesv1alpha1.CnsCsiSvFeatureStates
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(newObj.(*unstructured.Unstructured).Object, &svFSSObject)
+	if err != nil {
+		log.Errorf("fssCRUpdated: failed to cast object to %s. err: %v", featurestates.CRDSingular, err)
+		return
+	}
+	if svFSSObject.Name != featurestates.SVFeatureStateCRName {
+		log.Warnf("fssCRUpdated: Ignoring %s CR object with name %q", featurestates.CRDSingular, svFSSObject.Name)
+		return
+	}
+	for _, fss := range svFSSObject.Spec.FeatureStates {
+		k8sOrchestratorInstance.supervisorFSS.featureStates[fss.Name] = strconv.FormatBool(fss.Enabled)
+	}
+	log.Warnf("fssCRUpdated: New supervisor feature states values stored successfully from %s CR object: %v",
+		featurestates.SVFeatureStateCRName, k8sOrchestratorInstance.supervisorFSS.featureStates)
+}
+
+// fssCRDeleted marks feature state switch values for all features in supervisor feature state
+// switch to false and logs an error
```

Comment:
Can we document that this function also crashes the container? Since we're crashing the container, do we need to set the values to false. It doesnt matter right?

Reply:
Changed the docstring. It doesn't matter if we set the values to false or not, the container keeps crashing till we re-create the CR.